core/translate/emitter.rs

```diff
@@ -126,1642 +85,1661 @@ pub struct Metadata {
+            id,
+            table_reference,
+            search,
+            ..
+        } => {
+            let table_cursor_id = program.alloc_cursor_id(
+                Some(table_reference.table_identifier.clone()),
+                Some(Table::BTree(table_reference.table.clone())),
+            );
+
+            let next_row_label = program.allocate_label();
+
+            if !matches!(search, Search::PrimaryKeyEq { .. }) {
+                // Primary key equality search is handled with a SeekRowid instruction which does not loop, since it is a single row lookup.
+                m.next_row_labels.insert(*id, next_row_label);
+            }
 
-                            let agg_step_label = program.allocate_label();
+            let scan_loop_body_label = program.allocate_label();
+            m.scan_loop_body_labels.push(scan_loop_body_label);
+            program.emit_insn(Insn::OpenReadAsync {
+                cursor_id: table_cursor_id,
+                root_page: table_reference.table.root_page,
+            });
+            program.emit_insn(Insn::OpenReadAwait);
+
+            if let Search::IndexSearch { index, .. } = search {
+                let index_cursor_id = program
+                    .alloc_cursor_id(Some(index.name.clone()), Some(Table::Index(index.clone())));
+                program.emit_insn(Insn::OpenReadAsync {
+                    cursor_id: index_cursor_id,
+                    root_page: index.root_page,
+                });
+                program.emit_insn(Insn::OpenReadAwait);
+            }
 
-                            program.add_comment(
-                                program.offset(),
-                                "start new group if comparison is not equal",
-                            );
-                            program.emit_insn_with_label_dependency(
-                                Insn::Jump {
-                                    target_pc_lt: program.offset() + 1,
-                                    target_pc_eq: agg_step_label,
-                                    target_pc_gt: program.offset() + 1,
-                                },
-                                agg_step_label,
-                            );
+            return Ok(());
+        }
+        SourceOperator::Nothing => {
+            return Ok(());
+        }
+    }
+}
 
-                            program.emit_insn(Insn::Move {
-                                source_reg: groups_start_reg,
-                                dest_reg: comparison_register,
-                                count: group_by.len(),
-                            });
+/// Set up the main query execution loop
+/// For example in the case of a nested table scan, this means emitting the RewindAsync instruction
+/// for all tables involved, outermost first.
+fn open_loop(
+    program: &mut ProgramBuilder,
+    source: &mut SourceOperator,
+    referenced_tables: &[BTreeTableReference],
+    m: &mut Metadata,
```

Comment:
```suggestion
    metadata: &mut Metadata,
```

Reply:
changed m->metadata everywhere in https://github.com/penberg/limbo/pull/416/commits/db462530f1969b500b97dabdf3845f9ef730625a